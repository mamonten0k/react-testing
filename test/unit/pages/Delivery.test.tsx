import React from "react";
import { it } from "@jest/globals";
import { renderRoute, renderFragment } from "../render";

import { Delivery } from "../../../src/client/pages/Delivery";

describe("<Delivery>", () => {
  it("Представляет собой статичный контент", () => {
    const { asFragment } = renderFragment(<Delivery />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("По умолчанию присвоен роут /delivery", () => {
    const { container } = renderRoute("/delivery");
    expect(container.firstChild).toHaveClass("Delivery");
  });
});
