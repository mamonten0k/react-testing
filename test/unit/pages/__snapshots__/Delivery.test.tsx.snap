// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Delivery> Представляет собой статичный контент 1`] = `
<DocumentFragment>
  <div
    class="Delivery"
  >
    <div
      class="row"
    >
      <div
        class="col"
      >
        <h1>
          Delivery
        </h1>
        <p>
          Deserunt occaecati tempora. Qui occaecati est aliquam. Enim qui nulla ipsam. Incidunt impedit enim consequuntur amet at consequuntur vero. Dolor et ad facere asperiores iste est praesentium quaerat iure. Quibusdam mollitia autem quos voluptas quia est doloremque corporis et. Sed fuga quasi esse perspiciatis fugit maxime. Qui quidem amet.
        </p>
        <img
          class="Image w-25 mb-4"
          src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mNkMAYAADkANVKH3ScAAAAASUVORK5CYII="
        />
        <p>
          Dolores magnam consequatur iste aliquam qui sint non ab. Culpa saepe omnis. Recusandae vel aperiam voluptates harum. Perspiciatis qui molestiae qui tempora quisquam. Mollitia voluptatum minus laboriosam. Dolor maiores possimus repudiandae praesentium hic eos. Veritatis et repellat.
        </p>
        <p>
          Pariatur nisi nobis hic ut facilis sunt rerum id error. Soluta nihil quisquam quia rerum illo. Ipsam et suscipit est iure incidunt quasi et eum. Culpa libero dignissimos recusandae. In magni sapiente non voluptas molestias. Deserunt quos quo placeat sunt. Ea necessitatibus dolores eaque ex aperiam sunt eius. Saepe aperiam aut. Quaerat natus consequatur aut est id saepe et aut facilis.
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
